templates/hello/simple-snippet.html

```diff
@@ -0,0 +1,204 @@
+
+          Mozilla.UITour.observe = function(listener, callback) {
+            notificationListener = listener;
+
+            if (listener) {
+              document.addEventListener('mozUITourNotification',
+              _notificationListener);
+              Mozilla.UITour.ping(callback);
+            } else {
+              document.removeEventListener('mozUITourNotification',
+              _notificationListener);
+            }
+          };
+
+          Mozilla.UITour.registerPageID = function(pageID) {
+            _sendEvent('registerPageID', {
+              pageID: pageID
+            });
+          };
+
+          Mozilla.UITour.showMenu = function(name, callback) {
+            var showCallbackID;
+            if (callback)
+              showCallbackID = _waitForCallback(callback);
+
+            _sendEvent('showMenu', {
+              name: name,
+              showCallbackID: showCallbackID,
+            });
+          };
+
+          Mozilla.UITour.hideMenu = function(name) {
+            _sendEvent('hideMenu', {
+              name: name
+            });
+          };
+
+          Mozilla.UITour.getConfiguration = function(configName, callback) {
+            _sendEvent('getConfiguration', {
+              callbackID: _waitForCallback(callback),
+              configuration: configName,
+            });
+          };
+
+          Mozilla.UITour.trackHelloStatus = function () {
+            Mozilla.UITour.getConfiguration('availableTargets', function(config) {
+              if (config.targets && config.targets.indexOf('loop') > -1) {
+                sendMetric('hello-eligible-view');
+              } else {
+                sendMetric('hello-ineligible-view');
+              }
+            });
+          };
+
+          function isTargetAvailable (target, callback) {
+            Mozilla.UITour.getConfiguration('availableTargets', function(config) {
+              callback(config.targets && config.targets.indexOf(target) > -1);
+            });
+          }
+
+          function bindHelloObserver () {
+            Mozilla.UITour.observe(function(e) {
+              switch (e) {
+                case 'Loop:ChatWindowOpened':
+                  sendMetric('hello-start-conversation');
+                  break;
+                case 'Loop:RoomURLCopied':
+                  sendMetric('hello-url-copied');
+                  break;
+                // bug 1118346
+                case 'Loop:RoomURLEmailed':
+                  sendMetric('hello-url-emailed');
+                  break;
+              }
+            });
+          }
+
+         function handleVisibilityChange () {
+            if (document.hidden) {
+              // hide Hello menu & stop observer when changing tabs or minimizing window
+              Mozilla.UITour.observe(null);
+            } else {
+              // listen for Hello menu/chat window events
+              bindHelloObserver();
+            }
+          }
+
+          function thumbsUp () {
```

Comment:
`thumbsUp` is a weird name, doesn't really give a good idea of what this function does. I'd almost just take this and the contents of `bindHelloObserver` and put them directly in the code that calls these functions as they're not really so complex as to require separate functions, nor are they reused.

(Actually `bindHelloObserver` is reused but I have another comment about that coming.)